src/VersionParser.php

```diff
@@ -112,6 +121,20 @@ public function isDevelopment($version)
         return $this->development($version);
     }
 
+    /**
+     * Checks if two version strings are the same normalised version.
+     * 
+     * @param  string
+     * @param  string
+     * @return bool
+     */
+    public static function equals($version1, $version2)
+    {
+        $parser = new Parser;
+        return $parser->normalize(self::stripGitHash($version1))
+            === $parser->normalize(self::stripGitHash($version2)); 
+    }
```

Comment:
`Composer\Semver` does not take kindly to versions containing git metadata tagged on to the end of it. This can be injected by Box, for example, when building PHARs. We use a small method to strip it out before passing to Semver.